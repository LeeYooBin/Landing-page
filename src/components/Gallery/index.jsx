import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';
import theme from '../../styles/theme';

const Content = styled.div`
  max-width: 100%;
  height: 27rem;
  // padding: 0 0.5rem;
  display: flex;
  align-items: center;
  gap: 1rem;
  overflow-x: scroll;

  &::-webkit-scrollbar {
    height: 0.5rem;
    background-color: ${theme.colors.gray4};
  }

  &::-webkit-scrollbar-thumb {
    background-color: ${theme.colors.sun};
    border-radius: 0.5rem;
  }
`;

const Image = styled.img`
  width: 100%;
  height: 25rem;
  border-radius: 0.8rem;

  @media screen and (min-width: 920px) {
    width: 32rem;
  }
`;
const Gallery = ({ imageList }) => (
  <Content>
    {imageList.map((item) => (
      <Image key={uuidv4()} src={item.image} alt={item.alt} />
    ))}
  </Content>
);

Gallery.propTypes = {
  imageList: PropTypes.arrayOf(
    PropTypes.shape({
      image: PropTypes.string.isRequired,
      alt: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default Gallery;
