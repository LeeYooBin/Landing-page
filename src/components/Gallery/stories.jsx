import Gallery from '.';

export default {
  title: 'Gallery',
  component: Gallery,
};

const imageList = [
  {
    image:
      'https://static.vecteezy.com/ti/fotos-gratis/p3/6671766-fantastica-lua-magica-luz-e-agua-barco-com-arvore-papel-de-parede-gratis-foto.jpg',
    alt: 'image',
  },
  {
    image:
      'https://static.vecteezy.com/ti/fotos-gratis/p3/6671766-fantastica-lua-magica-luz-e-agua-barco-com-arvore-papel-de-parede-gratis-foto.jpg',
    alt: 'image',
  },
  {
    image:
      'https://static.vecteezy.com/ti/fotos-gratis/p3/6671766-fantastica-lua-magica-luz-e-agua-barco-com-arvore-papel-de-parede-gratis-foto.jpg',
    alt: 'image',
  },
  {
    image:
      'https://static.vecteezy.com/ti/fotos-gratis/p3/6671766-fantastica-lua-magica-luz-e-agua-barco-com-arvore-papel-de-parede-gratis-foto.jpg',
    alt: 'image',
  },
];

export const Template = (args) => <Gallery imageList={imageList} {...args} />;
