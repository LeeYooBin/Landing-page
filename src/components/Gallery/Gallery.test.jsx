import { describe, it, expect } from '@jest/globals';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import Gallery from '.';
import renderTheme from '../../styles/render-theme';

const mockImageList = [
  {
    image: 'image1.jpg',
    alt: 'Image 1',
  },
  {
    image: 'image2.jpg',
    alt: 'Image 2',
  },
];

describe('Heading component', () => {
  it('renders without errors', () => {
    renderTheme(<Gallery imageList={mockImageList} />);
  });

  it('renders correct number of images', () => {
    const { container } = renderTheme(<Gallery imageList={mockImageList} />);
    const images = container.querySelectorAll('img');
    expect(images.length).toBe(mockImageList.length);
  });

  it('renders images with correct src and alt attributes', () => {
    const { container } = renderTheme(<Gallery imageList={mockImageList} />);
    const images = container.querySelectorAll('img');

    images.forEach((img, index) => {
      expect(img.getAttribute('src')).toBe(mockImageList[index].image);
      expect(img.getAttribute('alt')).toBe(mockImageList[index].alt);
    });
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(<Gallery imageList={mockImageList} />);
    expect(container).toMatchSnapshot();
  });
});
