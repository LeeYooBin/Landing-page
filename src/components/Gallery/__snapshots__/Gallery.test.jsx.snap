// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component Should match snapshot 1`] = `
.c0 {
  max-width: 100%;
  height: 27rem;
  display: flex;
  align-items: center;
  gap: 1rem;
  overflow-x: scroll;
}

.c0::-webkit-scrollbar {
  height: 0.5rem;
  background-color: #898A93;
}

.c0::-webkit-scrollbar-thumb {
  background-color: #F5D15F;
  border-radius: 0.5rem;
}

.c1 {
  width: 100%;
  height: 25rem;
  border-radius: 0.8rem;
}

@media screen and (min-width: 920px) {
  .c1 {
    width: 32rem;
  }
}

<div>
  <div
    class="c0"
  >
    <img
      alt="Image 1"
      class="c1"
      src="image1.jpg"
    />
    <img
      alt="Image 2"
      class="c1"
      src="image2.jpg"
    />
  </div>
</div>
`;
